Core/source/core/pgp-key.ts

```diff
@@ -135,6 +135,14 @@ export class PgpKey {
     return p instanceof SecretKeyPacket || p instanceof SecretSubkeyPacket;
   }
 
+  public static validateAllDecryptedPackets = async (key: Key): Promise<void> => {
+    for (const prvPacket of key.toPacketList()) {
+      if (prvPacket instanceof SecretKeyPacket && prvPacket.isDecrypted()) {
```

Comment:
What about `SecretSubkeyPacket`? Why not just use `isPacketPrivate()` for filtering? Something like:
```typescript
for (const packet of key.toPacketList().filter(PgpKey.isPacketPrivate)) {
  await (packet as BaseSecretKeyPacket).validate();
}
```